src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,107 @@
+package duke;
+
+import duke.data.Calender;
+import duke.exception.BadDateException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the storage file for DukeProject.
+ */
+public class Storage {
+    private File file;
+
+    /**
+     * Prints the file in storage into an ArrayList.
+     * Takes in a calender to update the calender with the tasks in the file.
+     * @param calender calender to be updated by tasks in storage file.
+     * @return ArrayList made from tasks in storage file.
+     */
+    public ArrayList<Task> printFileIntoList(Calender calender) {
+        try {
+            Scanner sc = new Scanner(file);
+            ArrayList<Task> tasks = new ArrayList<>();
+            while (sc.hasNext()) {
+                String str = sc.nextLine();
+                String[] parts = str.split("\\|");
+                if (parts[0].equals("T")) {
+                    Task task = new Todo(Boolean.parseBoolean(parts[1]), parts[2]);
+                    tasks.add(task);
+                    calender.addDate(task);
+                } else if (parts[0].equals("E")) {
+                    Task task = new Event(Boolean.parseBoolean(parts[1]), parts[2], parts[3]);
+                    tasks.add(task);
+                    calender.addDate(task);
+                } else if (parts[0].equals("D")) {
+                    Task task = new Deadline(Boolean.parseBoolean(parts[1]), parts[2], parts[3]);
+                    tasks.add(task);
+                    calender.addDate(task);
+                }
+            }
+            return tasks;
+        } catch (FileNotFoundException e) {
+            System.err.println("File not found");
+            return null;
+        } catch (BadDateException e) {
+            System.err.println("Bad dates in File\n Please correct it and reload");
+            return null;
+        }
+    }
+
+    /**
+     * Writes the tasks in an ArrayList into the storage file.
+     * @param tasks ArrayList of tasks to be written into the storage file.
+     */
+    public void writeListIntoFile(ArrayList<Task> tasks) {
+        try {
+            FileWriter fw = new FileWriter(file);
+            for (Task task : tasks) {
+                switch (task.getType()) {
+                case TODO:
+                    fw.write("T|" + task.getDone() + "|" + task.getTask()
```

Comment:
Perhaps define helper methods for these, ie naming them like .writeToDo("String s") or similar for better readability.